src/core/stream/representation/representation_stream.ts

```diff
@@ -274,79 +256,60 @@ export default function RepresentationStream<T>({
+                 initSegmentState.segment !== null)
       {
         // prepend initialization segment
         const initSegmentPriority = neededSegments[0].priority;
-        neededSegments.unshift({ segment: initSegment,
-                                 priority: initSegmentPriority });
+        neededInitSegment = { segment: initSegmentState.segment,
+                              priority: initSegmentPriority };
       }
 
       const mostNeededSegment = neededSegments[0];
+      const initSegmentRequest = downloadingQueue.getCurrentInitRequest();
+      const currentSegmentRequest = downloadingQueue.getCurrentMediaRequest();
 
-      if (terminate !== null) {
-        downloadQueue = [];
-        if (terminate.urgent) {
-          log.debug("Stream: urgent termination request, terminate.", bufferType);
-          startDownloadingQueue$.complete(); // complete the downloading queue
-          return observableOf(EVENTS.streamTerminating());
-        } else if (currentSegmentRequest === null) {
-          log.debug("Stream: no request, terminate.", bufferType);
-          startDownloadingQueue$.complete(); // complete the downloading queue
-          return observableOf(EVENTS.streamTerminating());
-        } else if (
-          mostNeededSegment === undefined ||
-          currentSegmentRequest.segment.id !== mostNeededSegment.segment.id
-        ) {
-          log.debug("Stream: cancel request and terminate.", bufferType);
-          startDownloadingQueue$.next(); // interrupt the current request
-          startDownloadingQueue$.complete(); // complete the downloading queue
+      if (terminate === null) {
+        downloadQueue$.next({ initSegment: neededInitSegment,
+                              segmentQueue: neededSegments });
+      } else {
+        const { urgent } = terminate;
+        if (urgent) {
+          log.debug("Stream: Urgent switch, terminate now.", bufferType);
+          downloadQueue$.next({ initSegment: null, segmentQueue: [] });
+          downloadQueue$.complete();
           return observableOf(EVENTS.streamTerminating());
-        } else if (currentSegmentRequest.priority !== mostNeededSegment.priority) {
-          const { request$ } = currentSegmentRequest;
-          currentSegmentRequest.priority = mostNeededSegment.priority;
-          segmentFetcher.updatePriority(request$, mostNeededSegment.priority);
-        }
-        log.debug("Stream: terminate after request.", bufferType);
-      } else if (mostNeededSegment === undefined) {
-        if (currentSegmentRequest !== null) {
-          log.debug("Stream: interrupt segment request.", bufferType);
+        } else {
+          const nextQueue = currentSegmentRequest === null ||
+                            mostNeededSegment === undefined ||
+                            currentSegmentRequest.id !== mostNeededSegment.segment.id ?
+            [] :
+            [mostNeededSegment];
+
+          const nextInit = initSegmentRequest === null ? null :
```

Comment:
Is equivalent to :
 const nextInit = initSegmentRequest ?? neededInitSegment; (as initSegmentRequest canno't be null)
 As you want.

Reply:
>as initSegmentRequest canno't be null

What? I think you made a mistake but I still don't understand what you tried to say.

Reply:
Can't be `undefined` *